From 8978ff30b03e141b5a8578a19ba5fb3b25073faa Mon Sep 17 00:00:00 2001
From: Todd Whiteman <todd.whiteman@joyent.com>
Date: Mon, 18 Jun 2018 13:59:21 -0700
Subject: [PATCH] TRITON-518 sdc-docker service crash on createVolume with
 missing network Reviewed by: Orlando Vazquez <orlando@joyent.com> Approved
 by: Orlando Vazquez <orlando@joyent.com>

---
 lib/backends/sdc/volumes.js | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/lib/backends/sdc/volumes.js b/lib/backends/sdc/volumes.js
index 2655e80..065efc7 100644
--- a/lib/backends/sdc/volumes.js
+++ b/lib/backends/sdc/volumes.js
@@ -5,7 +5,7 @@
  */
 
 /*
- * Copyright (c) 2017, Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 var assert = require('assert-plus');
@@ -126,7 +126,9 @@ function createVolume(volumeParams, options, callback) {
                 mod_networks.findNetworkOrPoolByNameOrId(volumeParams.network,
                     networkOpts,
                     function onGetNamedNetwork(getNetworkErr, network) {
-                        payload.networks = [network.uuid];
+                        if (!getNetworkErr) {
+                            payload.networks = [network.uuid];
+                        }
                         next(getNetworkErr);
                     });
             }
-- 
2.21.0

